docs/DeveloperGuide.md

```diff
@@ -1,25 +1,290 @@
+
+## 3. Design & Implementation
+This section describes how features/functions are implemented.
+
+### Add Function
+
+#### System Architecture of the Add Function<br/>
+
+![](images/Architect-digram.PNG)
+
+The Architecture Diagram given above explains the high-level design of the add command.<br/>
+
+Below we will discuss a quick overview of each component.
+
+* UI: The User Interface of the app<br/>
+* Add Command: The main logic command of the add function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+
+#### Design of the Add Function<br/>
+
+UML Class Diagram of the Add function:
+![](images/uml.PNG)<br/>
+
+1) The Add function gets the inputs from the User and places it to the event class so that 
+it can parser to string.<br/>
+2)  It will then pass the event data onto UI class to display and inform user of the event added 
+to the timetable.<br/>
+3) Next,the add function will pass all the parameters (event name,day,time
+and location) to the timetable class into its arraylist divided by the day of the week<br/>
+
+#### Sequence Design of Add Function<br/>
+![](images/seqdiagram.jpg)<br/>
+
+#### Improvements to be made to Version 2.0<br/>
+* To ensure that the users do not enter duplicate event that has the same timing or in between the time 
+that is entered. 
+* A verification check is added to the add command of the version 2.0 of WhereGotTime.
+
+E.g.
+* add /CS2113 Lec /Mon /1200-1400 /LT21<br/>
+* add /CS2113 Lec /Mon /1300-1400 /LT22<br/>
+* add /CS2040C Tut /Mon /1300-1500 /LT32<br/>
+
+The above would result in the first line being add,
+while the other two will result in the app giving an error message
+prompting the user to re-enter again.<br/>
+
+### Edit Function<br/>
+
+#### Class Diagram <br/>
+
+![](images/ClassDiagram_EditCommand.png)<br/>
+
+### Delete Function
+
+#### System Architecture of the Delete Function<br/>
+
+![](images/DeleteCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the delete command.<br/>
+* UI: The User Interface of the app<br/>
+* DeleteCommand: The main logic command of the delete function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Delete Function<br/>
+
+![](images/DeleteCommand_umlDiagram.png)
+
+1. `DeleteCommand` gets the input from the User and gets the relevant timetable from `Timetable`
+1. `DeleteCommand` then proceeds to delete the event in the timetable based on the index in the user input
+1. `DeleteCommand` passes the day, deleted event and size of timetable to the `UI`
+1. `UI` updates the user of the successful deletion of the event
+
+#### Sequence Design of Delete Function<br/>
+![](images/DeleteCommand_SequenceDiagram.png)<br/>
+
+### Clear Function
+
+#### System Architecture of the Clear Function<br/>
```

Comment:
Perhaps the system architecture of the delete and clear functions could be consolidated, since they are similar in nature?